client.py

```diff
@@ -12,72 +12,82 @@
-    page1.on("framenavigated", lambda frame: print("Frame navigated to %s" % frame.url))
-    page1.on("request", lambda request: print("Request %s" % request.url))
-    page1.on(
-        "requestFinished", lambda request: print("Request finished %s" % request.url)
-    )
-    page1.on(
-        "response",
-        lambda response: print(
-            "Response %s, request %s in frame %s"
-            % (response.url, response.request.url, response.frame.url)
-        ),
+def main() -> None:
+    browser = playwright_sync.chromium.launch(headless=False)
+    page = browser.newPage(viewport=0)
+    page.setContent(
+        "<button id=button onclick=\"window.open('http://webkit.org', '_blank')\">Click me</input>"
     )
-    print("Navigating page1 to https://example.com...")
-    await page1.goto("https://example.com")
-    print("Page1 main frame url: %s" % page1.mainFrame.url)
-    print("Page1 tile: %s" % await page1.title())
-    print("Frames in page1: %d" % len(page1.frames))
-    await page1.screenshot(path="example.png")
 
-    print("\nCreating page2...")
-    page2 = await context.newPage()
-    page2.on("framenavigated", lambda frame: print("Frame navigated to %s" % frame.url))
+    with page.expect_event("popup") as popup:
+        page.click("#button")
+    print(popup.data)
+    sleep(100)
+
+    # print("Contexts in browser: %d" % len(browser.contexts))
+    # print("Creating context...")
+    # context = browser.newContext(viewport=0)
+    # print("Contexts in browser: %d" % len(browser.contexts))
+    # print("Pages in context: %d" % len(context.pages))
+
+    # print("\nCreating page1...")
+    # page1 = context.newPage()
+    # print("Pages in context: %d" % len(context.pages))
+    # page1.on("framenavigated", lambda frame: print("Frame navigated to %s" % frame.url))
+    # page1.on("request", lambda request: print("Request %s" % request.url))
+    # page1.on(
+    #     "requestFinished", lambda request: print("Request finished %s" % request.url)
+    # )
+    # page1.on(
+    #     "response",
+    #     lambda response: print(
+    #         "Response %s, request %s in frame %s"
+    #         % (response.url, response.request.url, response.frame.url)
+    #     ),
+    # )
+    # print("Navigating page1 to https://example.com...")
+    # page1.goto("https://example.com")
+    # print("Page1 main frame url: %s" % page1.mainFrame.url)
+    # print("Page1 tile: %s" % page1.title())
+    # print("Frames in page1: %d" % len(page1.frames))
+    # page1.screenshot(path="example.png")
+
+    # print("\nCreating page2...")
+    # page2 = context.newPage()
+    # page2.on("framenavigated", lambda frame: print("Frame navigated to %s" % frame.url))
 
-    print("Navigating page2 to https://webkit.org...")
-    await page2.goto("https://webkit.org")
-    print("Page2 tile: %s" % await page2.title())
-    print("Pages in context: %d" % len(context.pages))
+    # print("Navigating page2 to https://webkit.org...")
```

Comment:
commit commented code is not that nice, lets have maybe both versions in it by a cli argument?

Reply:
Oops, done